ERCS/erc-7772.md

```diff
@@ -0,0 +1,271 @@
+---
+eip: 7772
+title: Partial Gas Sponsorship
+description: Defining an interface that dApps can use to sponsor a portion of the gas fees required for user operations
+author: Lyu Min (@rockmin216), Wu Jinzhou (@wujinzhou), Elwin Chua (@jingkang0822), Lucas Lim (@limyeechern)
+discussions-to: https://ethereum-magicians.org/t/erc-7772-partial-gas-sponsorship-interface/21145
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-09-20
+requires: 4337
+---
+
+## Abstract
+This proposal defines the necessary interface that decentralized applications (dApps) must implement to sponsor a portion of the required gas for user operations utilizing a Paymaster that supports this standard. The proposal also provides a suggested code implementation that Paymasters can include in their current implementation to support dApp sponsorship. Partial sponsorship between more than one dApps may also be achieved through this proposal.
+
+## Motivation
+This proposal introduces a standard that enables dApps to sponsor a portion of these transaction fees on behalf of users. Currently, TokenPaymasters allow users to pay gas fees using ERC-20 tokens instead of blockchain's native token. However, this approach does not equate to TokenPaymasters directly sponsoring the transaction fees, as users still bear the cost when the ERC-20 token is transferred to TokenPaymaster. 
+
+Similarly, with a Free-Gas Paymaster, the cost of the entire UserOperation is borne by the Paymaster, and therefore may not be willing to sponsor some transaction. With this proposal, dApps can sponsor a portion of the transaction fees and therefore, by reducing the cost of interacting with dApps, this standard incentivizes increased user activity and attracts more users to the dApp.
+
+From the user's perspective, this reduction in transaction costs encourages greater participation in the blockchain ecosystem, as interacting with dApps becomes more affordable. Additionally, with IGasSponsor, partial sponsorship can be achieved when a UserOperation involves multiple dApps that implement IGasSponsor within a single transaction, further enhancing the user experience.
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Contract Interface
+This interface is designed to be implemented by dApps that want to manage their gas funds internally and allow Paymasters to claim gas costs in a standardised way.
+
+``` solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the document must be `CC0-1.0` licensed. Code in your `assets/` folder can be `MIT` if you'd like.